### ISSUE TITLE
Bug: indent/keyword-spacing errors with hermes-eslint

### ISSUE THREAD

**Issue by Brianzchen** [eslint/eslint#17430]
### Environment

Node version: v16.15.0
npm version: v8.5.5
Local ESLint version: v8.45.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 22.5.0

### What parser are you using?

Other

### What did you do?

In this repo https://github.com/Brianzchen/hermes-eslint-repro I'm using eslint parser as https://www.npmjs.com/package/hermes-eslint but for the rules `keyword-spacing` and `indent` I'm getting getting errors with the syntax like

```js
export * as routes from './routes';
```

### What did you expect to happen?

It should parse and not error on the rules `indent` or `keyword-spacing`

### What actually happened?

```
➜  hermes-eslint-repro git:(master) yarn eslint .
yarn run v1.22.19
$ /Users/brianchen/projects/hermes-eslint-repro/node_modules/.bin/eslint .

Oops! Something went wrong! :(

ESLint: 8.45.0

TypeError: Cannot read properties of undefined (reading '0')
Occurred while linting /Users/brianchen/projects/hermes-eslint-repro/src/index.js:2
Rule: "keyword-spacing"
    at SourceCode.getTokenBefore (/Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/source-code/token-store/index.js:298:23)
    at checkSpacingForModuleDeclaration (/Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/rules/keyword-spacing.js:469:44)
    at ruleErrorHandler (/Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/linter/linter.js:1050:28)
    at /Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/Users/brianchen/projects/hermes-eslint-repro/node_modules/.pnpm/eslint@8.45.0/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Link to Minimal Reproducible Example

https://github.com/Brianzchen/hermes-eslint-repro

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I've reached out to the hermes team but hermes seems to be able to parse the code and it's the lint rule itself that fails to read the ast output of the parser. Hopefully you can provide a fix 🙏 

If I disable these 2 rules everything works as expected.

**Comment by Rec0iL99** [comment]
Hi @Brianzchen, thanks for the issue. I'm able to reproduce the error. I'm going to push this forward for the core team.

**Comment by mdjermanovic** [comment]
This is a problem in the `hermes-eslint` parser. The `routes` Identifier node is missing `range` property:

```js
const hermesESLint = require("hermes-eslint");

const { ast: program } = hermesESLint.parseForESLint(
    "// @flow\nexport * as routes from './routes';\n",
    {
        loc: true,
        range: true,
        raw: true,
        tokens: true,
        comment: true,
        eslintVisitorKeys: true,
        eslintScopeManager: true
    }
);

const { type, name, range } = program.body[0].exported;

console.log(type, name, range); // Identifier routes undefined
```

**Comment by Brianzchen** [comment]
Thanks for the feedback and debugging! I've managed to get it fixed on the parser end https://github.com/facebook/hermes/commit/9d5629762764c660ed3e96c5e64cc0f0e97f70cf
